### DISCUSSION TITLE
Boxer charging light flashing

### DISCUSSION THREAD

**Discussion by KennyCroft** [EdgeTX/edgetx#5479] [General]
Radiomaster Boxer Crush, ExpressLRS version internal module. Edge 2.10.2. ExpressLRS 3,4.3
Sometimes after an overnight charge of the 6200mah internal battery, using a wall-wart charger, the charge indication light is flashing on the Boxer transmitter. When the charging lead is removed the flashing stops and all is normal and the battery voltage shows as 8.3v.
Normally after charging fully the light simply goes out. Whilst charging the light is red and normally goes out when fully charged.
Why does this flashing sometimes happen.

**Comment by inventor7777** [comment]
I have the same behavior on MT12

**Reply by KennyCroft** [reply]
How often does it happen. Mine has happened just twice in a couple of weeks with the boxer.

**Reply by inventor7777** [reply]
It happens to me pretty much every time since I updated a couple months ago.

**Comment by pfeerick** [comment]
I believe flashing is an indication from the charge controller chip that it
is having trouble doing the top end balancing, and has timed out. It would
probably be a good idea to put the battery on an external charger or
battery checker to see what is going on with the cells. But often all you
need to do is put it on charge again, as it usually just  a little further
out of balance than the charger chip was able to deal with within its
charge cycle.

---------- Forwarded message ---------
From: KennyCroft ***@***.***>
Date: Tue, 27 Aug 2024, 6:23 am
Subject: Re: [EdgeTX/edgetx] Boxer charging light flashing (Discussion
#5479)
To: EdgeTX/edgetx ***@***.***>
Cc: Subscribed ***@***.***>


How often does it happen. Mine has happened just twice in a couple of weeks
with the boxer.

—
Reply to this email directly, view it on GitHub
<https://github.com/EdgeTX/edgetx/discussions/5479#discussioncomment-10455950>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ABJ66KPSNXP2ASRYBUBMOHTZTOFETAVCNFSM6AAAAABNCPNWPCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBVGU4TKMA>
.
You are receiving this because you are subscribed to this thread.Message
ID: ***@***.***>

**Reply by inventor7777** [reply]
Sounds good.

**Comment by KennyCroft** [comment]
Next time it happens I will check the voltage of the individual cells after
charging. If they are out of balance I will use an external charger
together them balanced.

**Comment by TGS78** [comment]
I had similar issue with my new Jumper T20 V2 two times now.
After inserting two brand new Samsung 5Ah 21700 Li-Ion cells, and charging over night while radio powered off, in the morning only 5 of 6 LEDs were on. EdgeTX showed 8.0V. Measurement with external charger gave me 4.1V on first and 3,9V on second battery. Seems the balancer circuit is "weak" and **_my assumption_** is that the battery circuit stopped charging, while MCU maybe is still running and slowly discharging the batteries again.

Another strange behaviour is that the power button RGB LED is always solid blue when charger is plugged in.
No matter if I'm charging, or battery is already fully charged, or even if no battery inserted at all.

This is contrary to the user manual
Charging indicator:
the green light flashes when the radio is not inserted with batteries;
the green light is always on when the battery is charging;
the green light is off when the charging is complete.

Is the power button RGB LED controlled by EdgeTX? Is the battery charging circuit connected to the main MCU? I there any signal/link between them?

**Comment by pfeerick** [comment]
Charging is always done by dedicated charge chip, nothing to do with
EdgeTX. RGB power led is mostly controlled by EdgeTX, but some handsets the
charge chip has direct control of one or more colours to indicate charge
status, so is up to manufacturer what happens there (colour and any
flashing sequences).

The 6 customisable switch led charge status is only a rough indicator of
charge status, as it does not take into account voltage calibration... Only
boots up the radio enough to measure the battery and run the LEDs.
